frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -820,16 +848,42 @@ function ReaderHighlight:updateHighlight(page, index, side, direction, move_by_c
     UIManager:setDirty(self.dialog, "ui")
 end
 
-function ReaderHighlight:onShowHighlightNoteOrDialog(page, index)
-    if self.select_mode then
-        if page ~= self.highlight_page or index ~= self.highlight_idx then return end
-        -- tap on the first fragment: abort select mode, clear highlight
-        self.select_mode = false
-        self:deleteHighlight(page, index)
-        return true
+function ReaderHighlight:showChooseHighlightDialog(highlights)
+    if #highlights == 1 then
```

Comment:
(The name of this method is about what happens in the `else` branch - so it may better have a more generic name. But can't really think of any, a generic `showHighlightsDialog()` is a bit misleading too. So, ok :)